_posts/2019-05-28-improved-hive-bucketing.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: post
+title:  "Improved Hive Bucketing"
+author: David Phillips
+---
+
+The next release of Presto (version 312) adds support for the more
+flexible bucketing introduced in recent versions of Hive. Specifically,
+it allows any number of files per bucket, including zero. This allows
+inserting data into an existing partition without having to rewrite the
+entire partition, and improves the performance of writes by not requiring
+the creation of files for empty buckets.
+
+# Hive bucketing overview
+
+Hive bucketing is a simple form of hash partitioning. A table is bucketed
+on one or more columns with a fixed number of hash buckets:
+
+```sql
+CREATE TABLE page_views (
+  user_id bigint,
+  page_url varchar,
+  dt date
+)
+WITH (
+  partitioned_by = ARRAY['dt'],
+  bucketed_by = ARRAY['user_id'],
```

Comment:
You describe "Hive bucketing" but provide Presto DDL. Would be good to make it explicit. eg "For example, a table definition in Presto may look like this"